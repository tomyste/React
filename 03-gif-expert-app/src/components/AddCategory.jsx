import  PropTypes  from "prop-types";
import { useState } from "react"


export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) =>{
        setInputValue(target.value)
    }

    const onSubmit = (event) =>{
      
        // el metodo event.prevent default permite que no se refresque la pagina cada vez que le damos enter al input
        event.preventDefault();

        // Evalua si existe algo en la caja de texto, de lo contrario no hace nada
        if(inputValue.trim().length <= 1) return ;

        // El trim() se encarga de cortar los espacios en blancos adelante y atras del texto
        onNewCategory(inputValue.trim())

        // Resetea el value de la caja de texto
        setInputValue('')
    }

  return (
    <form onSubmit={ onSubmit } aria-label='form'>
        <input 
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
        >
        </input>
    </form>
    
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}