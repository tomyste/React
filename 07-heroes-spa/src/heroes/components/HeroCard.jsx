import React from 'react'
import { Link } from 'react-router-dom';
import { heroes } from '../data/heroes'

export const HeroCard = ({id, superhero, publisher, alter_ego, first_appearance, characters }) => {

    const heroImageUrl = `/assets/heroes/${ id }.jpg`;

    



  return (
    <div className='col animate__animated animate__fadeIn'>
        <div className='card'>
            <div className='row no-gutters'>
                <div className='col-5'>
                    <img src={heroImageUrl} className='card-img' alt={superhero}></img>
                </div>

                <div className='card-body col-5'>
                    <h5 className='card-title'> { superhero } </h5>
                    <p className='card-text'> { alter_ego }</p>

                    {
                        ( alter_ego !== characters )&& (<p> { characters }</p>)
                    }
                    <p className='card-text'>
                        <small className='text-muted'>
                            {first_appearance}
                        </small>
                    </p>
                    
                    
                    <Link id='Link' className='text-decoration-none' to={`/hero/${id}`}> Mas...</Link>
                    
                    
                    
                </div>

            </div>
        </div>
    </div>
  )
}
