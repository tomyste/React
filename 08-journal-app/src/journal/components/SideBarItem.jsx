import { TurnedInNot } from "@mui/icons-material"
import { Grid, ListItem, ListItemButton, ListItemIcon, ListItemText } from "@mui/material"
import { useMemo } from "react";
import { useDispatch } from "react-redux";
import { setActiveNote } from "../../store/journal";




export const SideBarItem = ({ title = '', id, body , date, imageUrls = []}) => {

    const dispatch = useDispatch();    

    const newTitle = useMemo(() => {
        return title.length > 15 
        ? title.substring(0,15) + '...'
        : title;
    }, [ title ])

    const newBody = useMemo(() => {
        return body.length > 20 
        ? body.substring(0,20) + '...'
        : body;
    }, [ body ])

    const onClickNote = () => {
        dispatch(setActiveNote({title, id, body, date, imageUrls }))

        
    }

  

  return (
    <ListItem disablePadding>
        <ListItemButton onClick={onClickNote}>
            <ListItemIcon>
                <TurnedInNot />
            </ListItemIcon>
            <Grid container>
                <ListItemText primary={ newTitle } />
                <ListItemText secondary={ newBody } />
            </Grid>
        </ListItemButton>
    </ListItem>
  )
}
