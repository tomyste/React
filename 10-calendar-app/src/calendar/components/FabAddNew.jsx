import { addHours } from 'date-fns';
import React from 'react'
import { useDispatch } from 'react-redux'
import { useCalendar, useUiStore } from '../../hooks';

export const FabAddNew = () => {
    const { openDateModal } = useUiStore();
    const { setActiveEvent } = useCalendar();

    const handleClickNew = () => {
        setActiveEvent(
            { 
                title: '',
                notes: '',
                start: new Date(),
                end: addHours( new Date(), 2 ),
                bgColor: '#fafafa',
                user: {
                  _id: '12412',
                  name: 'tomys'
                }
              }
        )
        openDateModal();
    }

  return (
    <button
    className='btn btn-primary fab'
    onClick={ handleClickNew }
    >
      <i className='fas fa-plus'></i>
    </button>
  )
}
