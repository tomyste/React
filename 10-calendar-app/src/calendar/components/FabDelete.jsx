
import React from 'react'
import { useCalendar, useUiStore } from '../../hooks';


export const FabDelete= () => {
    const { startDeletingEvent, hasEventSelected } = useCalendar();
    const { isDateModalOpen } = useUiStore();

    const handleDelete = () => {
      startDeletingEvent();
    }

  return (
    <button
    className='btn btn-danger fab-danger'
    onClick={ handleDelete }
    style={{
      display: hasEventSelected && !isDateModalOpen ? '' : 'none',
    }}
    >
        <i className='fas fa-trash-alt'></i>
    </button>
  )
}
