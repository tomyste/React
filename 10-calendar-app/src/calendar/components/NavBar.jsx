

export const NavBar = () => {
  return (
    <div className="navbar navbar-dar bg-dark mb-4 px-4">
      <span className="navbar-brand">
        <i className="fas fa-calendar-alt"/>
        &nbsp;
        Tomas
      </span>

      <button className="btn btn-outline-danger">
        <i className="fas fa-sign-out-alt"/>
        &nbsp;
        <span>Salir</span>
      </button>
    </div>
  )
}
