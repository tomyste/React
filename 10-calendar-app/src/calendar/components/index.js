export * from './NavBar'
export * from './CalendarEventBox'
export * from './CalendarModal'
export * from './FabAddNew'