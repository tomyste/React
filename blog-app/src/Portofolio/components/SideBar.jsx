import "../cssStyles/sideBar.css"
import { SideBarItem } from "./SideBarItem"
import HomeIcon from '@mui/icons-material/Home';
import { BookmarkBorder, MessageSharp, NotificationsNone, PermIdentity, Search } from "@mui/icons-material";
import { Button, Drawer, styled, Tooltip, tooltipClasses, Typography } from "@mui/material";
import { Box } from "@mui/system";

//TODO: Ajustar medidas: maximo ->  595 y minimo: (VER),  medio tambien. Averiguar como hacer para que sea con movimiento fluido



export const SideBar = () => {


  return (
    <Box className="sidebar"
    component='nav'
    sx={{ width: {sm: 595}, flexShrink: {
        sm: 0
    }}}    
    >
        <Drawer
            variant='permanent' // temporary
            open
            sx={{ 
                display: { xs: 'block' },
                '& .MuiDrawer-paper': { boxSizing: 'border-box', width: {xl: 595, sm: 150, md: 300}, alignItems: 'center', mt: 8 }
            }}
        >


        <SideBarItem text={ "Home" } Icon={ HomeIcon }/>
        <SideBarItem text={ "Explore" } Icon={ Search }/>
        <SideBarItem text={ "Notifications" } Icon={ NotificationsNone }/>
        <SideBarItem text={ "Bookmarks" } Icon={ BookmarkBorder }/>
        <SideBarItem text={ "Messages" } Icon={ MessageSharp }/>
        <SideBarItem text={ "Profile" } Icon={ PermIdentity }/>     
        <Button className="button" id="newPost"
        style={{ width: '50%' }}
        sx={{mt: {xl: 70, md:60, sm: 40}}}> NewPost </Button> 
        </Drawer>

        
    </Box>
  )
}
