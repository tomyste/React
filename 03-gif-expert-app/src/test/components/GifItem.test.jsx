import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";

describe('Pruebas en GifItem', () => {
    const title = 'Dragon Ball';
    const url = 'https://one-punch.com/saitama.jpg'
    
    test('Debe de hacer match con el snapshot', () => { 
        
        const {container} = render(<GifItem title={ title } url={ url }></GifItem>);
        // screen.debug(); -> Si quisiera mostrar en la consola los resultados del screen
        expect( container ).toMatchSnapshot();
        
    })


    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        render(<GifItem title={ title } url={ url }></GifItem>);
        
        const { src, alt}  = screen.getByRole('img');

        expect ( src ).toBe(  url );
        expect ( alt ).toBe ( alt )
     })


     test('Debe de mostrar el titulo en el componente', () => { 
        
        render(<GifItem title={ title } url={ url }></GifItem>);
        expect(screen.getByText( title )).toBeTruthy;
      })
});
