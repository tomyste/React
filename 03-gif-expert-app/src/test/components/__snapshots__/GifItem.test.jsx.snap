// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Dragon Ball
    </p>
  </div>
</div>
`;
